modules/ROOT/pages/traversal-framework/traversal-framework-java-api.adoc

```diff
@@ -357,20 +357,20 @@ class MaxWeightPathExpander implements PathExpander<Double>
     }
 
     @Override
-    public Iterable<Relationship> expand( Path path, BranchState<Double> branchState )
+    public ResourceIterable<Relationship> expand( Path path, BranchState<Double> branchState )
     {
         if (path.lastRelationship() != null) {
             branchState.setState( branchState.getState() + (double) path.lastRelationship().getProperty( "weight" ) );
         }
 
-        Iterable<Relationship> relationships = path.endNode().getRelationships( Direction.OUTGOING );
+        ResourceIterable<Relationship> relationships = path.endNode().getRelationships( Direction.OUTGOING );
         ArrayList<Relationship> filtered = new ArrayList<>();
         for ( Relationship relationship : relationships ) {
             if ( branchState.getState() + (double) relationship.getProperty( "weight" ) <= maxWeight ) {
                 filtered.add(relationship);
             }
         }
-        return filtered;
+        return Iterables.asResourceIterable(filtered);
```

Comment:
I've accepted the change, though it does use an internal method (`Iterables` utility class is not part of the public api)
We don't have an alternative currently that I am aware of, other than the user writing their own wrapper.

Reply:
I believe we have informed customers who do use the traversal framework to use this method in the past.